galsim/chromatic.py

```diff
@@ -227,6 +460,152 @@ def drawImage(self, bandpass, image=None, integrator=None, **kwargs):
         image += integral
         return image
 
+    def _interp_drawImage(self, bandpass, image=None, integrator=None, **kwargs):
+        """Draw method adapted to work for ChromaticImage instances for which interpolation between
+        stored images is being used.  Users should not call this routine directly, and should
+        instead interact with the `drawImage` method.
+        """
+        # Check whether the object has had some chromatic transformation applied to it.  Since we
+        # are interpolating between stored images, this is not a situation that _interp_drawImage
+        # can handle.  We have to lose the interpolation.
+        if hasattr(self, '_A') and self._chromaticTransformation(bandpass):
```

Comment:
I was making the opposite point actually, since I didn't notice that `_chromaticTransformation` also checked for a chromatic flux scaling.  So I thought you were neglecting to remove the interpolation in that case, which would be an error.

Since the code you have is correct, I don't see a huge need to make the optimization you mentioned.  The user should normally be able to optimize this themselves by waiting until after they apply their chromatic flux scaling before calling `setupInterpolation`.  And the warning will tell them they ought to.